src/app/tracking/tracking.directive.ts

```diff
@@ -0,0 +1,39 @@
+import {Directive, HostListener} from '@angular/core';
+import {AngularFireDatabase} from 'angularfire2/database';
+import {AngularFireAuth} from 'angularfire2/auth';
+import {Router} from '@angular/router';
+import {PresentationComponent} from '../presentation/presentation/presentation.component';
+
+@Directive({
+  selector: '[app-tracking]'
+})
+export class TrackingDirective {
+  auth;
+  lastSlideChange;
+  history: Array<any> = [];
+
+  constructor(private afDb: AngularFireDatabase, private afAuth: AngularFireAuth,
+              private router: Router, private presentation: PresentationComponent) {
+    afAuth.auth.signInAnonymously();
+    afAuth.authState.subscribe(authData => {
+      this.auth = authData;
+    });
+    this.lastSlideChange = Date.now();
+  }
+
+  @HostListener('onSlideChange', ['$event']) onSlideChange(index) {
+    if (this.auth) {
+      const diffMinutes = Date.now() - this.lastSlideChange;
```

Comment:
You're calling `Date.now()` three times in this method. Consider adding a const and using it three times.